packages/backend/src/tokens/types.ts

```diff
@@ -10,5 +10,60 @@ export type AuthenticateRequestOptions = {
   signUpUrl?: string;
   afterSignInUrl?: string;
   afterSignUpUrl?: string;
+  organizationSyncOptions?: OrganizationSyncOptions;
   apiClient?: ApiClient;
 } & VerifyTokenOptions;
+
+/**
+ * Defines the options for syncing an organization or personal workspace state from the URL to the clerk session.
+ * Useful if the application requires the inclusion of a URL that indicates that a given clerk organization
+ * (or personal workspace) must be active on the clerk session.
+ *
+ * If a mismatch between the active organization on the session and the organization as indicated by the URL is
+ * detected, an attempt to activate the given organization will be made.
+ *
+ * WARNING: If the activation cannot be performed, either because an organization does not exist or the user lacks
+ * access, then the active organization on the session will not be changed (and a warning will be logged). It is
+ * ultimately the responsibility of the page to verify that the resources are appropriate to render given the URL,
+ * and handle mismatches appropriately (e.g. by returning a 404).
+ */
+export type OrganizationSyncOptions = {
+  /**
+   * URL patterns that are organization-specific and contain an organization ID or slug as a path token.
+   * If a request matches this path, the organization identifier will be extracted and activated before rendering.
+   *
+   * WARNING: If the organization cannot be activated either because it does not exist or the user lacks access,
+   * organization-related fields will be set to null. The server component must detect this and respond
+   * with an appropriate error (e.g., notFound()).
+   *
+   * If the route also matches the personalWorkspacePatterns, the personalWorkspacePattern takes precedence.
+   *
+   * Must have a path token named either ":id" (matches a clerk organization ID) or ":slug" (matches a clerk
+   * organization slug).
+   *
+   * Common examples:
+   * - ["/orgs/:slug", "/orgs/:slug/(.*)"]
+   * - ["/orgs/:id", "/orgs/:id/(.*)"]
+   * - ["/app/:any/orgs/:slug", "/app/:any/orgs/:slug/(.*)"]
+   */
+  organizationPatterns?: Array<Pattern>;
```

Comment:
I _think_ this is generally the canonical syntax when defining an array type:

```suggestion
  organizationPatterns?: Pattern[];
```

There's no functional difference either way, this is just a shorthand for `Array<Pattern>`

Reply:
👍  4076c031d66ac56955f56e8b44dff0db2d7af6b9